src/main/python/main/ayab/preferences.py

```diff
@@ -226,7 +226,7 @@ def __init__(self, parent: GuiMain):
         for widget in self.__widget.values():
             widget.connectChange()
         self.__ui.reset.clicked.connect(self.__reset_and_refresh)
-        self.__ui.enter.clicked.connect(self.accept)
+        self.__ui.enter.clicked.connect(self.__validate_form)
```

Comment:
The way the Preferences dialog is currently setup, the `OK` button is a lie. Each setting is updated as soon as you change its value, event though nothing happens immediately in the main window because the main event loop is blocked by the popup. You can verify this by closing the Preferences dialog without clicking `OK`, i.e. press Escape or click the window close button. Demonstration:

https://github.com/user-attachments/assets/970f7fde-77dd-4134-9bc1-0e202594be91

Therefore, intercepting the click on the `OK` button would make the UI refresh only for users that click it, causing weird behaviors.

The Preferences dialog should probably be reworked to actually commit settings only when `OK` is clicked, but that's a bigger piece of work.

Reply:
Ohhh yes, I missed that :( I'll try to find an other solution.

Reply:
@jonathanperret the done method seems to be triggered in all cases that allow to close the window, (ESC, close button, OK button). Do you see something else ?

Reply:
For anyone reading this later, [my later comment](https://github.com/AllYarnsAreBeautiful/ayab-desktop/pull/700#issuecomment-2308759538) and @mathPi 's following changes obviate this discussion.